### PR TITLE
pkg/volume/util/operationexecutor:remove prometheus dependencies from it

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
 
/kind cleanup
 
**What this PR does / why we need it**:

As part of Metrics Stability KEP we migrated from Prometheus client to Kubernetes Metrics Framework. So we no longer need to depend on promethues client directly.

/cc @serathius
/cc  @RainbowMango 

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/89267

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Same as https://github.com/kubernetes/kubernetes/pull/89285#pullrequestreview-378296807.

File: pkg/volume/util/operationexecutor/operation_generator_test.go [hunk 1]
```diff
@@ -89,12 +88,12 @@ func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
 			"volume_plugin":  expectedPluginName,
 		}
 		// compare the relative change of the metric because of the global state of the prometheus.DefaultGatherer.Gather()
```
Comment (tanjunchen) [inline pkg/volume/util/operationexecutor/operation_generator_test.go]: /cc @RainbowMango  
hello , i have some trouble by ussing  *AndCompare etc here.
here , we need to use them ? could you give me some help.Thank you very much.

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 2]
```diff
@@ -193,6 +193,59 @@ func ValidateMetrics(metrics Metrics, metricName string, expectedLabels ...strin
...
+func getMetricFamily(metricFamily []*dto.MetricFamily, metricFamilyName string) *dto.MetricFamily {
+	for _, mf := range metricFamily {
+		if mf.GetName() == metricFamilyName {
+			return mf
+		}
+	}
+	return nil
+}
+
+// LabelsMatch returns true if metric has all expected labels otherwise false
```
Comment (tanjunchen) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:227]: moving there , reference code as close as possible

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 3]
```diff
@@ -193,6 +193,59 @@ func ValidateMetrics(metrics Metrics, metricName string, expectedLabels ...strin
...
+func LabelsMatch(metric *dto.Metric, labelFilter map[string]string) bool {
+	metricLabels := map[string]string{}
```
Comment (tanjunchen) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:229]: Reference code as close as possible

File: pkg/volume/util/operationexecutor/operation_generator_test.go [hunk 4]
```diff
@@ -89,57 +90,46 @@ func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
...
+			metricValueDiff := storageOperationStatusCountMetricAfterValue - storageOperationStatusCountMetricBeforeValue
 			assert.Equal(t, float64(1), metricValueDiff, tc.name)
 		}
 	}
 }
 
-func findMetricWithNameAndLabels(metricFamilyName string, labelFilter map[string]string) *io_prometheus_client.Metric {
-	metricFamily := getMetricFamily(metricFamilyName)
+// getMetricCounterValueWithNameAndLabels returns metric value according to metricFamilyName and labelFilter
+func getMetricCounterValueWithNameAndLabels(metricFamilyName string, labelFilter map[string]string) (float64, error) {
```
Comment (tanjunchen) [inline pkg/volume/util/operationexecutor/operation_generator_test.go:113]: /cc @logicalhan 
how about this ?  We just need to extract the common function GetMetricFamily() in metric .

**Pull request closed without merging**